timbuctoo-instancev4/src/test/java/nl/knaw/huygens/timbuctoo/v5/datastores/rssource/ChangeListBuilderTest.java

```diff
@@ -35,23 +37,72 @@ public void retrieveChangeFilesNamesReturnsCorrectNamesBasedOnSuppliedVersions()
-    given(cursorQuad.getDirection()).willReturn(Direction.OUT);
-
-    given(bdbTruePatchStore.getChangesOfVersion(1, true)).willReturn(Stream.of(cursorQuad));
+    CursorQuad cursorQuad1 = mock(CursorQuad.class);
+    given(cursorQuad1.getSubject()).willReturn("s1");
+    given(cursorQuad1.getPredicate()).willReturn("p1");
+    given(cursorQuad1.getChangeType()).willReturn(ChangeType.ASSERTED);
+    given(cursorQuad1.getObject()).willReturn("o1");
+    given(cursorQuad1.getDirection()).willReturn(Direction.OUT);
+
+    CursorQuad cursorQuad2 = mock(CursorQuad.class);
+    given(cursorQuad2.getSubject()).willReturn("s2");
+    given(cursorQuad2.getPredicate()).willReturn("p2");
+    given(cursorQuad2.getChangeType()).willReturn(ChangeType.RETRACTED);
+    given(cursorQuad2.getObject()).willReturn("o2");
+    given(cursorQuad2.getDirection()).willReturn(Direction.OUT);
+
+    CursorQuad cursorQuad3 = mock(CursorQuad.class);
+    given(cursorQuad3.getSubject()).willReturn("s3");
+    given(cursorQuad3.getPredicate()).willReturn("p3");
+    given(cursorQuad3.getChangeType()).willReturn(ChangeType.ASSERTED);
+    given(cursorQuad3.getObject()).willReturn("o3");
+    given(cursorQuad3.getDirection()).willReturn(Direction.OUT);
+    given(cursorQuad3.getValuetype()).willReturn(Optional.of(RdfConstants.STRING));
+
+    CursorQuad cursorQuad4 = mock(CursorQuad.class);
+    given(cursorQuad4.getSubject()).willReturn("s4");
+    given(cursorQuad4.getPredicate()).willReturn("p4");
+    given(cursorQuad4.getChangeType()).willReturn(ChangeType.RETRACTED);
+    given(cursorQuad4.getObject()).willReturn("o4");
+    given(cursorQuad4.getDirection()).willReturn(Direction.OUT);
+    given(cursorQuad4.getValuetype()).willReturn(Optional.of(RdfConstants.STRING));
+
+    CursorQuad cursorQuad5 = mock(CursorQuad.class);
+    given(cursorQuad5.getSubject()).willReturn("s5");
+    given(cursorQuad5.getPredicate()).willReturn("p5");
+    given(cursorQuad5.getChangeType()).willReturn(ChangeType.ASSERTED);
+    given(cursorQuad5.getObject()).willReturn("o5");
+    given(cursorQuad5.getDirection()).willReturn(Direction.OUT);
+    given(cursorQuad5.getValuetype()).willReturn(Optional.of(RdfConstants.LANGSTRING));
+    given(cursorQuad5.getLanguage()).willReturn(Optional.of("en"));
+
+    CursorQuad cursorQuad6 = mock(CursorQuad.class);
+    given(cursorQuad6.getSubject()).willReturn("s6");
+    given(cursorQuad6.getPredicate()).willReturn("p6");
+    given(cursorQuad6.getChangeType()).willReturn(ChangeType.RETRACTED);
+    given(cursorQuad6.getObject()).willReturn("o6");
+    given(cursorQuad6.getDirection()).willReturn(Direction.OUT);
+    given(cursorQuad6.getValuetype()).willReturn(Optional.of(RdfConstants.LANGSTRING));
+    given(cursorQuad6.getLanguage()).willReturn(Optional.of("en"));
+
+    given(bdbTruePatchStore.getChangesOfVersion(1, true))
+      .willReturn(Stream.of(cursorQuad1, cursorQuad2, cursorQuad3, cursorQuad4, cursorQuad5, cursorQuad6));
```

Comment:
Mock returns a retraction too (cursorQuad2 is a retraction), though _getChangesOfVersion(1, true)_ should only return assertions. A real instance of the BdbTruePatchStore for this test will solve false assumptions like this.

Reply:
Now uses a non persistent database and BdbTruePatchStore is no longer mocked.